.ipynb_checkpoints/BW2_tutorial-checkpoint.ipynb

```diff
@@ -0,0 +1,5219 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "imp = bw.ExcelImporter(os.path.join(\"excel_importer_example.xlsx\"))\n",
+    "imp.apply_strategies()\n",
+    "imp.match_database(fields=('name', 'unit', 'location'))\n",
+    "imp.match_database(\"ecoinvent 3.5_cutoff_ecoSpold02\", fields=('name', 'unit', 'location'))\n",
+    "imp.statistics()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can check whether the import went as expected by having a look at an Excel sheet, that includes our process data. The location of this file is given as output of the following command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imp.write_excel()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Seems like everything went fine :)\n",
+    "\n",
+    "If an exchange cannot be linked using the above applied strategies, because the respective input is listed multiple times in the database, you would have to link it manually. An example for such a case is `treatment of aluminium scrap, post-consumer, prepared for recycling, at refiner`, for which two ecoinvent (3.5 cutoff) entries exist."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Having imported the data, we also need to write it to a database to save it:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "imp.write_database()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bw.databases"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wbi = bw.Database(\"BW2 Excel water bottle import\")\n",
+    "for act in wbi:\n",
+    "    print(act)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wbp = [act for act in wbi][0]\n",
+    "for exc in wbp.exchanges():\n",
+    "    print(exc)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "See whether the details of our biosphere exchange are correct:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ex = [act for act in wbi if 'water bottle production' in act['name']][0]\n",
+    "[exc for exc in ex.exchanges() if 'Carbon dioxide' in str(exc)][0].as_dict()"
```

Comment:
----> 4 [exc for exc in ex.exchanges() if 'Carbon dioxide' in str(exc)][0].as_dict()

IndexError: list index out of range